crates/core/src/factories/core_fetch.rs

```diff
@@ -156,23 +207,26 @@ struct State {
 #[derive(Debug)]
 struct CoreFetch {
     state: Arc<Mutex<State>>,
-    fetch_queue_tx: Sender<FetchRequest>,
-    fetch_tasks: Vec<JoinHandle<()>>,
+    request_tx: Sender<FetchRequest>,
+    tasks: Vec<JoinHandle<()>>,
+    #[cfg(test)]
+    response_handler: DynTxModuleHandler,
```

Comment:
I'm not so happy with this, but somehow I need to send messages to the response queue. This seemed the simplest.

Reply:
meh, I'm fine with this : )